vehicles/Hero/main.cc

```diff
@@ -0,0 +1,828 @@
+        // break condition (loading)
+        if (++i > 10 && abs(load_servo->GetOmega()) <= 0.001 && !GimbalDead) break;
+        // loading once
+        if (i == 1) {
+          Preload = false;
+          load_servo->SetTarget(load_servo->GetTarget() - load_angle);
+        }
+        load_servo->CalcOutput();
+        control::MotorCANBase::TransmitOutput(can2_loader, 1);
+        osDelay(SHOOTER_INNER_TASK_DELAY);
+      }
+      // after loading
+      i = 0;
+      load_motor->SetOutput(0);
+      control::MotorCANBase::TransmitOutput(can2_loader, 1);
+      osDelay(100); // need test the delay time
+    }
+
+    if (dbus->swl == remote::DOWN) {
+      force_weak = false;
+    }
+
+    // whether change the force motor position
+    ForceWeakDetect.input(dbus->keyboard.bit.F || dbus->wheel.wheel < 1024);
+    ForceStrongDetect.input(dbus->keyboard.bit.C
+                          || (dbus->wheel.wheel == remote::WheelDigitalValue && dbus->previous_wheel_value == remote::WheelDigitalValue));
+    // just execute force transform once
+    if (ForceWeakDetect.posEdge() && !ForceStrongDetect.posEdge() && !force_weak) {
+      force_transforming = true;
+      force_weak = true;
+      force_strong = false;
+    } else if (ForceStrongDetect.posEdge() && !ForceWeakDetect.posEdge() && !force_strong) {
+      force_transforming = true;
+      force_weak = false;
+      force_strong = true;
+    }
+    // force transforming
+    if (force_transforming && !GimbalDead) {
+      while (true) {
+        // break condition (reach the desire position)
+        if (++i > 30 && abs(force_motor->GetOmega()) <= 0.001 && !GimbalDead) break;
+        // set the speed and acceleration for the reload motor
+        // set target pull position once
+        if (i == 1) {
+          // direction need test
+          if (force_weak && !force_strong) {
+            force_pos += force_angle;
+          } else if (force_strong && !force_weak) {
+            force_pos -= force_angle;
+          }
+          force_servo->SetTarget(force_pos);
+        }
+        force_servo->CalcOutput();
+        control::MotorCANBase::TransmitOutput(can2_force, 1);
+        osDelay(SHOOTER_INNER_TASK_DELAY);
+      }
+      // after changing the force motor position
+      i = 0;
+      force_transforming = false;
+      force_motor->SetOutput(0);
+      control::MotorCANBase::TransmitOutput(can2_force, 1);
+      osDelay(SHOOTER_OUTER_TASK_DELAY); // need test the delay time(wait for the)
+    }
+
+    // the shoot and load process is automatic
+    // 1. using servo pull the trigger (shooting process)
+    // 2. reload motor pull the bullet board to the desire position (load process below)
+    // 3. using servo push the trigger the hold the bullet board
+    // 4. load motor load one bullet on the board
+    // 5. reload motor release to the original position to prepare for the next load
+
+    // Load Detector
+    LoadDetect.input(dbus->swr == remote::UP || dbus->mouse.l);
```

Comment:
why called "LoadDetect" should it not be "Trigger detect“？

Reply:
just detect the load signal.